processor/src/advice/dsa.rs

```diff
@@ -0,0 +1,68 @@
+use super::{ExecutionError, Felt, StarkField, Vec, Word};
+use vm_core::{
+    crypto::dsa::rpo_falcon512::{KeyPair, Polynomial},
+    utils::Deserializable,
+};
+
+/// Gets as input a vector containing an expanded public key and its associated secret key, and a
+/// word representing a message and outputs a vector of values to be pushed onto the advice stack.
+/// The values are the ones required for a Falcon signature verification inside the VM and they are:
+///
+/// 1. The nonce represented as 8 field elements.
+/// 2. The expanded public key represented as the coefficients of a polynomial of degree < 512.
+/// 3. The signature represented as the coefficients of a polynomial of degree < 512.
+/// 4. The product of the above two polynomials in the ring of polynomials with coefficients
+/// in the Miden field.
+///
+/// # Errors
+/// Will return an error if either:
+/// - The advice map does not contain an entry with key PK.
+/// - The advice map entry under key PK is not a vector of the expected length.
```

Comment:
I would probably replace the second point here with something more general (i.e., the error will be returned not only if the length is off, but also if the keys are generally malformed in some other ways). Also, maybe worth adding a 3rd point about failing to generate a signature - but maybe that's the same as the 2nd point.

Reply:
Actually both points are out of place. I have updated both. Let me know if it makes sense.